core/include/scipp/core/variable.h

```diff
@@ -25,8 +25,8 @@ struct SCIPP_CORE_EXPORT Slice {
   Slice(const Dim dim_, const scipp::index begin_, const scipp::index end_ = -1)
       : dim(dim_), begin(begin_), end(end_) {}
   Dim dim;
-  scipp::index begin;
-  scipp::index end;
+  const scipp::index begin;
+  const scipp::index end;
```

Comment:
Why this as const but not `dim`? What is the reason behind making it const? Doesn't this prevent assignment (not sure that is relevant)?

Reply:
This is the remanence  of looking at `Slice` to see if any of the basic validation could be put there. The default value for the `end` index seemed too problematic at present, but I thought that if we were to improve this in future, making the `Slice` immutable would be a useful step. It turns out that all access via the members is read-only at present.